model/model.go

```diff
@@ -230,11 +231,101 @@ type TableInfo struct {
 	Compression string `json:"compression"`
 
 	View *ViewInfo `json:"view"`
+	// Lock represent the table lock info.
+	Lock *TableLockInfo `json:"Lock"`
 
 	// Version means the version of the table info.
 	Version uint16 `json:"version"`
 }
 
+// TableLockInfo provides meta data describing a table lock.
+type TableLockInfo struct {
+	Tp TableLockType
+	// Use array because there maybe multiple session both hold same read lock.
+	Sessions []SessionInfo
+	State    TableLockState
+	// TS use to record the timestamp of this table lock been public locked.
+	TS uint64
+}
+
+// SessionInfo contain the session ID and the server ID.
+type SessionInfo struct {
+	ServerID  string
+	SessionID uint64
+}
+
+func (s SessionInfo) String() string {
+	return "server:" + s.ServerID + "_session:" + strconv.FormatUint(s.SessionID, 10)
+}
+
+// DBTableID is composed by schema ID and table ID.
+type DBTableID struct {
+	DBID    int64
+	TableID int64
+}
+
+// TableLockTpInfo is composed by schema ID, table ID and table lock type.
+type TableLockTpInfo struct {
+	SchemaID int64
+	TableID  int64
+	Tp       TableLockType
+}
+
+// TableLockState is the state for table lock.
+type TableLockState byte
+
+const (
+	// TableLockStateNone means this table lock is absent.
+	TableLockStateNone TableLockState = iota
+	// TableLockStatePreLock means this table lock is pre-lock state. Other session doesn't hold this lock should't do corresponding operation according to the lock type.
+	TableLockStatePreLock
+	// TableLockStatePublic means this table lock is public state.
+	TableLockStatePublic
+)
+
+// String implements fmt.Stringer interface.
+func (t TableLockState) String() string {
+	switch t {
+	case TableLockStatePreLock:
+		return "pre-lock"
+	case TableLockStatePublic:
+		return "public"
+	default:
+		return "none"
+	}
+}
+
+// TableLockType is the type of the table lock.
+type TableLockType byte
+
+const (
+	TableLockNone TableLockType = iota
+	// TableLockRead means the session with this lock has read permission, other session can't read/write until the lock is released.
```

Comment:
IMHO we need to add more detailed explanations. For example, if the locks are exclusive, or can be shared.

Although the behavior of `LOCKS` will be explained in our official document, detailed comments are still useful here for the developers.

Reply:
I've checked the implementations, this comment is wrong(`other session can't read/write ...`), please fix it.

Reply:
Sorry for my mistakes, PTAL again.